six/modules/c++/samples/geolocate_pixel.cpp

```diff
@@ -0,0 +1,120 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <exception>
+#include <iostream>
+
+#include <cli/ArgumentParser.h>
+#include <cli/Results.h>
+#include <import/except.h>
+#include <import/six/sicd.h>
+#include "utils.h"
+
+
+int main(int argc, char** argv)
+{
+    try
+    {
+        cli::ArgumentParser parser;
+        parser.setDescription("Given a SICD, and a pixel in the corresponding "
+                              "output plane, display the lat/lon/alt");
+        parser.addArgument("-s --schemaPath",
+                           "Specify a schema or directory of schemata",
+                           cli::STORE, "schema", "FILE");
+        parser.addArgument("-r --rotateToShadowsDown",
+                           "Specify whether to rotate to shadows-down",
+                           cli::STORE, "shadowsDown", "SHADOWS")->setDefault(true);
+        parser.addArgument("input", "Input SICD/SIDD file", cli::STORE, "input",
+                           "INPUT", 1, 1);
+        parser.addArgument("row", "Pixel row location", cli::STORE, "row",
+                           "ROW", 1, 1);
+        parser.addArgument("col", "Pixel col location", cli::STORE, "col",
+                           "COL", 1, 1);
+        const std::auto_ptr<cli::Results>
+            options(parser.parse(argc, (const char**) argv));
+
+        sys::OS os;
+        const std::string sicdPathname(options->get<std::string>("input"));
+        if (!os.exists(sicdPathname))
+        {
+            std::cerr << sicdPathname << " does not exist.\n";
+            return 1;
+        }
+
+        std::vector<std::string> schemaPaths;
+        getSchemaPaths(*options, "--schemaPath", "schema", schemaPaths);
+
+        const double row = options->get<double>("row");
+        const double col = options->get<double>("col");
+        const six::RowColDouble rowCol(row, col);
+
+        six::XMLControlRegistry xmlRegistry;
+        xmlRegistry.addCreator(six::DataType::COMPLEX,
+                new six::XMLControlCreatorT<six::sicd::ComplexXMLControl>());
+
+        std::auto_ptr<six::sicd::ComplexData> complexData;
+        std::vector<std::complex<float> > widebandData;
+        six::sicd::Utilities::readSicd(sicdPathname, schemaPaths, complexData,
+                widebandData);
+
+        const bool shadowsDown(options->get<bool>("shadowsDown"));
+        if (!six::sicd::AreaPlaneUtility::hasAreaPlane(*complexData))
+        {
+            six::sicd::AreaPlaneUtility::setAreaPlane(*complexData);
+        }
+
+        six::sicd::AreaPlane& plane = *complexData->radarCollection->area->plane;
+        if (shadowsDown)
+        {
+            plane.rotateCCW();
```

Comment:
You can't just always rotate CCW if you want shadows down... you need to check the orientation to see if you should rotate.  Probably worth adding a `rotateToShadowsDown()` method to the plane that does this check for you.  Note that if the orientation is `ARBITRARY`, you won't want to rotate it at all